app/channel-settings/server/methods/saveRoomSettings.js

```diff
@@ -243,6 +236,16 @@ Meteor.methods({
 			});
 		}
 
+		if (!hasPermission(userId, 'edit-room', rid)) {
+			if (!(Object.keys(settings).includes('encrypted') && room.t === 'd')) {
+				throw new Meteor.Error('error-action-not-allowed', 'Editing room is not allowed', {
+					method: 'saveRoomSettings',
+					action: 'Editing_room',
+				});
+			}
+			settings = { encrypted: settings.encrypted };
```

Comment:
Hey @MartinSchoeler , I just stumbled upon this line in the code and can't figure out why it exists. Do you have any context?